### DISCUSSION TITLE
当设备横屏的时候，字体问题

### DISCUSSION THREAD

**Discussion by pisyell** [NervJS/taro#13913] [Q & A]
h5，做了一个横屏预览的功能，与app通信使其横屏的时候，字体会突然变大
```
@media screen and (orientation: landscape) {
    html,
    body {
        font-size:  ???; // 这里如何使用竖屏的时候的字体大小，或者有没有其他解决方案
    }
}
```

**Comment by GauharChan** [comment]
同样求助 @Chen-jj @ZakaryCode 
原因是window resize之后根节点的font-size更新了，有没有什么方式解决？

**Comment by yoyo837** [comment]
这不是就是内置的rem功能吗? https://taro-docs.jd.com/docs/config-detail#designwidth 有关的

**Comment by GauharChan** [comment]
> 这不是就是内置的rem功能吗? https://taro-docs.jd.com/docs/config-detail#designwidth 有关的

我的问题是兼容竖屏和横屏，有没有什么方法控制根元素的font-size保持一致。类似使用vmin的效果

**Comment by yoyo837** [comment]
rem本身就是基于宽度的, 宽度变化字体自然就会变了.

**Comment by yoyo837** [comment]
> > 这不是就是内置的rem功能吗? https://taro-docs.jd.com/docs/config-detail#designwidth 有关的
> 
> 我的问题是兼容竖屏和横屏，有没有什么方法控制根元素的font-size保持一致。类似使用vmin的效果

那只能改实现了, 取最小值来判断

**Comment by GauharChan** [comment]
> > > 这不是就是内置的rem功能吗? https://taro-docs.jd.com/docs/config-detail#designwidth 有关的
> > 
> > 
> > 我的问题是兼容竖屏和横屏，有没有什么方法控制根元素的font-size保持一致。类似使用vmin的效果
> 
> 那只能改实现了, 取最小值来判断

taro用的这个[postcss-pxtorem](https://github.com/cuth/postcss-pxtorem/)不支持vmin，并且这个插件已是5年没更新了

**Reply by yoyo837** [reply]
不是的, 用的 fork 的 `postcss-pxtransform`, 在 taro 仓库下.

**Reply by yoyo837** [reply]
https://github.com/NervJS/taro/pull/14703

**Reply by yoyo837** [reply]
组内讨论, 觉得你的建议不错, 准备改了. 也欢迎PR

**Comment by GauharChan** [comment]
> 组内讨论, 觉得你的建议不错, 准备改了. 也欢迎PR

ok，期待你们的更新

**Reply by yoyo837** [reply]
PR https://github.com/NervJS/taro/pull/14704
